hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -103,43 +101,80 @@ def get_splits(self) -> Tuple[TilesDataset, TilesDataset, TilesDataset]:
         """Create the training, validation, and test datasets"""
         raise NotImplementedError
 
-    def prepare_data(self) -> None:
-        if self.precache_location != CacheLocation.NONE:
-            self._load_dataset(self.train_dataset, stage='train', shuffle=True)
-            self._load_dataset(self.val_dataset, stage='val', shuffle=True)
-            self._load_dataset(self.test_dataset, stage='test', shuffle=True)
-
     def _dataset_pickle_path(self, stage: str) -> Optional[Path]:
         if self.cache_dir is None or self.cache_mode == CacheMode.NONE:
             return None
         return self.cache_dir / f"{stage}_dataset.pt"
 
+    def prepare_data(self) -> None:
```

Comment:
Better to raise an error saying this method needs to be implemented in the child class. It can help also to explain what th method should do given it returns None

Reply:
I think this is only needed in the TilesDataModule for caching, the SlidesDataModule does not implement it, I will remove it from the mother class